Add autoscale button to disable automatic y-axis scaling

davrsky opened the pull request:
Sometimes I don't want y-axis to automatically scale, adds a checkbox to enable/disable it

davrsky committed:
Add autoscale button to disable automatic y-axis scaling

dirk-thomas commented on src/rqt_plot/plot_widget.py:
```diff
@@ -154,6 +155,10 @@ def switch_data_plot_widget(self, data_plot):
         self.data_plot_layout.addWidget(self.data_plot)
         self.data_plot.autoscroll(self.autoscroll_checkbox.isChecked())
+        if self.autoscale_checkbox.isChecked():
+            self.data_plot.set_autoscale(y=DataPlot.SCALE_EXTEND|DataPlot.SCALE_VISIBLE)
```
The default value `SCALE_EXTEND|SCALE_VISIBLE` is currently being defined in the `plot.py` file. If the patch wants to make that toggle within the widget then the default value from the plot file should be removed and instead a function on the widget should be called to achieve the same behavior without duplicating these values.

davrsky committed:
Improve handling topic ambiguity

Given there are two topics:
/foo
/foo/bar

And the user requests to graph
/foo/bar/baz

Previously, it would randomly pick a topic (I think in practice it would
always pick /foo, but not sure). Now I make it pick the longest matching
topic, which is /foo/bar.

alex-makarov commented:
Merging this would boost the usability of rqt_plot *a lot*. Please merge..

lucbettaieb commented:
Please merge this!  `rqt_plot` gets into an absolutely unusable state very easily...

lucbettaieb commented:
@dirk-thomas <3

dirk-thomas commented:
@alex-makarov @lucbettaieb The pending comment stills needs to be resolved. Also there is a second completely unrelated commit on this PR which should be separate into its own PR. If you would like to see either of the two changes merged please consider to open PRs with the comments addressed.

lucbettaieb commented:
@dirk-thomas My bad!  I assumed the second commit contained the fixups.

sloretz commented:
Thank you for the PR!

Unfortunately I don't think we should merge this one.

[ROS Noetic will reach end-of-life on May 31st, 2025](https://discourse.ros.org/t/ros-noetic-end-of-life-may-31-2025/43160). Every change comes with a risk of introducing regressions, and there isn't much time left to fix them. I'm closing pull requests that add features so that the remaining time is allocated towards bug fixes and compatibility with newer Ubuntu distros.
